### PR TITLE
Submit trustedqsl to flathub

### PR DESCRIPTION
**Pull request by k1mu**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). - for this, I have ARRL permission to publish.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.arrl.trustedqsl.json [hunk 1]
```diff
@@ -0,0 +1,158 @@
+{
+    "app-id" : "org.arrl.trustedqsl",
+    "runtime" : "org.freedesktop.Platform",
+    "runtime-version" : "22.08",
+    "sdk" : "org.freedesktop.Sdk",
+    "sdk-extensions" : [ "org.freedesktop.Sdk.Locale" ],
```
Comment (hfiguiere) [inline org.arrl.trustedqsl.json:6]:
```suggestion
```

File: org.arrl.trustedqsl.json [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    "command" : "tqsl",
+    "finish-args" : [
+	"--share=network",
+	"--socket=x11",
```
Comment (hfiguiere) [inline org.arrl.trustedqsl.json:10]: `--share=ipc` is needed for x11

File: org.arrl.trustedqsl.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	"--filesystem=home"
+     ],
+    "modules" : [
+        {
+            "name" : "openssl",
```
Comment (hfiguiere) [inline org.arrl.trustedqsl.json:15]: openssl is in the runtime.

File: org.arrl.trustedqsl.json [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+		"/share/man/man1/*",
+		"/share/man/man1",
+		"/bin/mdb_copy",
+		"/bin/mdb_dump",
+		"/bin/mdb_load",
+		"/bin/mdb_stat"
+	    ]
+        },
+        {
+            "name" : "curl",
```
Comment (hfiguiere) [inline org.arrl.trustedqsl.json:73]: curl is in the the runtime.

File: org.arrl.trustedqsl.json [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                    "type" : "archive",
+                    "url" : "https://github.com/LMDB/lmdb/archive/LMDB_0.9.23.tar.gz",
+                    "sha256" : "abf42e91f046787ed642d9eb21812a5c473f3ba5854124484d16eadbe0aa9c81"
+                }
+            ],
+	    "cleanup" : [
+		"/include/lmdb.h",
+		"/lib/liblmdb.a",
+		"/share/man/man1/*",
+		"/share/man/man1",
```
Comment (hfiguiere) [inline org.arrl.trustedqsl.json:65]:
```suggestion
		"/include",
		"*.a",
		"/share/man",
```

File: org.arrl.trustedqsl.json [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+		"/share/man/man7",
+		"/share/aclocal/libcurl.m4",
+		"/share/aclocal",
+		"/lib/pkgconfig/libcurl.pc",
+		"/lib/pkgconfig"
+	    ]
+        },
+	{
+	    "name" : "wxWidgets",
+	    "buildsystem" : "cmake",
```
Comment (hfiguiere) [inline org.arrl.trustedqsl.json:105]: cmake-ninja should work.

File: org.arrl.trustedqsl.json [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+	    "name" : "tqsl",
+	    "buildsystem" : "cmake",
+    	    "config-opts" : [
+	        "-DTQSL_FLATPAK=Yes",
+		"-DCMAKE_BUILD_TYPE=Release"
+	    ],
+	    "sources" : [
+		{
+		    "type" : "archive",
+        	    "url" : "https://www.rickmurphy.net/lotw/tqsl-2.7.tar.gz",
```
Comment (hfiguiere) [inline org.arrl.trustedqsl.json:145]: is this a fork? This is not the same as found at https://www.arrl.org/tqsl/tqsl-2.6.5.tar.gz

File: org.arrl.trustedqsl.json [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        	    "sha256" : "128961d4a19f3e8b0144314b6c0d6900428cae12460b1cb2b4e373dccc9cfee7"
+	  	}
+	    ],
+	    "cleanup" : [
+		"/include/adif.h",
+		"/include/cabrillo.h",
+		"/include/tqslconvert.h",
+		"/include/tqslerrno.h",
+		"/include/tqsllib.h"
+	    ]
```
Comment (hfiguiere) [inline org.arrl.trustedqsl.json:155]:
```suggestion
	    "cleanup" : [
		"/include"
	    ]
```

File: org.arrl.trustedqsl.json [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+			"/bin/wx-config",
+			"/bin/wxrc",
+			"/bin/wxrc-3.2",
+			"/share/bakefile",
+			"/include/wx-3.2/wx/*",
+			"/include/wx-3.2/wx",
+			"/include/wx-3.2",
+			"/lib/cmake/wxWidgets/*",
+			"/lib/cmake/wxWidgets",
+			"/lib/cmmake"
```
Comment (hfiguiere) [inline org.arrl.trustedqsl.json:132]:
```suggestion
			"/lib/wx",
			"/bin/wx-config",
			"/bin/wxrc",
			"/bin/wxrc-3.2",
			"/share/bakefile",
			"/include",
			"/lib/cmake"
```

**Comment by hfiguiere** [comment]
bot, build org.arrl.trustedqsl

File: org.arrl.trustedqsl.json (hunk 2 above)
Comment (k1mu) [reply org.arrl.trustedqsl.json:10]: I'll add this and resubmit.

File: org.arrl.trustedqsl.json (hunk 3 above)
Comment (k1mu) [reply org.arrl.trustedqsl.json:15]: Openssl *development* is not in the runtime.

File: org.arrl.trustedqsl.json (hunk 4 above)
Comment (k1mu) [reply org.arrl.trustedqsl.json:73]: Curl *development* is not in the runtime.

File: org.arrl.trustedqsl.json (hunk 5 above)
Comment (k1mu) [reply org.arrl.trustedqsl.json:65]: change made. I'll verify that the man pages indeed get removed, and simplify the other manpage removals.

File: org.arrl.trustedqsl.json (hunk 7 above)
Comment (k1mu) [reply org.arrl.trustedqsl.json:145]: This is the current head of this package. I'll get pulled to arrl.org when it's released.

File: org.arrl.trustedqsl.json (hunk 8 above)
Comment (k1mu) [reply org.arrl.trustedqsl.json:155]: Change added.

File: org.arrl.trustedqsl.json (hunk 9 above)
Comment (k1mu) [reply org.arrl.trustedqsl.json:132]: Change made. I missed the typo in 'cmmake', thanks.

File: org.arrl.trustedqsl.json (hunk 3 above)
Comment (hfiguiere) [reply org.arrl.trustedqsl.json:14]: no package need to build openssl 3.

File: org.arrl.trustedqsl.json (hunk 4 above)
Comment (hfiguiere) [reply org.arrl.trustedqsl.json:71]: not package need to build curl.

File: org.arrl.trustedqsl.json (hunk 6 above)
Comment (k1mu) [reply org.arrl.trustedqsl.json:103]: Ok, perhaps naive question - is there an advantage of 'cmake-ninja' versus 'cmake'?

Comment (hfiguiere) [reply org.arrl.trustedqsl.json:103]: it's faster so it's prefered. Sometime it doesn't work though.

**Comment by k1mu** [comment]
Build failed, missing screenshot tags. 
I've run the "appinfo" validator and found several errors that need to be fixed:
/home/rmurphy/org.arrl.trustedqsl.xml: FAILED:
â¢ tag-missing           : org.arrl.trustedqsl: <category> must include main categories from the desktop entry spec
â¢ style-invalid         : org.arrl.trustedqsl: Not enough <screenshot> tags, minimum is 1
â¢ tag-missing           : org.arrl.trustedqsl: desktop application has no icon
Validation of files failed

**Comment by hfiguiere** [comment]
you do not need to close the PR to update it. This is just waste reviewer's time.

**Comment by k1mu** [comment]
Understood. Apologies from this newbie. :(

**Pull request closed without merging**